framework/src/headers/ip/v6/srh.rs

```diff
@@ -0,0 +1,481 @@
+    //                                                             //
+    //         Optional Type Length Value objects (variable)       //
+    //                                                             //
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+   where:
+
+   o  Next Header: 8-bit selector.  Identifies the type of header
+      immediately following the SRH.
+
+   o  Hdr Ext Len: 8-bit unsigned integer, is the length of the SRH
+      header in 8-octet units, not including the first 8 octets.
+
+   o  Routing Type: TBD, to be assigned by IANA (suggested value: 4).
+
+   o  Segments Left.  Defined in [RFC8200], it contains the index, in
+      the Segment List, of the next segment to inspect.  Segments Left
+      is decremented at each segment.
+
+   o  Last Entry: contains the index, in the Segment List, of the last
+      element of the Segment List.
+
+   o  Flags: 8 bits of flags.  Following flags are defined:
+
+          0 1 2 3 4 5 6 7
+         +-+-+-+-+-+-+-+-+
+         |U|P|O|A|H|  U  |
+         +-+-+-+-+-+-+-+-+
+
+         U: Unused and for future use.  SHOULD be unset on transmission
+         and MUST be ignored on receipt.
+
+         P-flag: Protected flag.  Set when the packet has been rerouted
+         through FRR mechanism by an SR endpoint node.
+
+         O-flag: OAM flag.  When set, it indicates that this packet is
+         an operations and management (OAM) packet.
+
+         A-flag: Alert flag.  If present, it means important Type Length
+         Value (TLV) objects are present.  See Section 3.1 for details
+         on TLVs objects.
+
+         H-flag: HMAC flag.  If set, the HMAC TLV is present and is
+         encoded as the last TLV of the SRH.  In other words, the last
+         36 octets of the SRH represent the HMAC information.  See
+         Section 3.1.5 for details on the HMAC TLV.
+
+   o  Tag: tag a packet as part of a class or group of packets, e.g.,
+      packets sharing the same set of properties.
+
+   o  Segment List[n]: 128 bit IPv6 addresses representing the nth
+      segment in the Segment List.  The Segment List is encoded starting
+      from the last segment of the path.  I.e., the first element of the
+      segment list (Segment List [0]) contains the last segment of the
+      path, the second element contains the penultimate segment of the
+      path and so on.
+
+   o  Type Length Value (TLV) are described in Section 3.1.
+*/
+
+// SRv6 Segment IDs are an array of 128-bit values with a length defined at
+// runtime.
+pub type Segments = [u128];
+
+// The v6 Segment Routing Header is a specialization of the v6 Routing Header,
+// defined in
+// https://tools.ietf.org/html/draft-ietf-6man-segment-routing-header-11. Like
+// all extension headers, it shares the first two fields.
+#[derive(Default)]
+#[repr(C, packed)]
+pub struct SegmentRoutingHeader<T>
+where
+    T: Ipv6VarHeader,
+{
+    ext_header: Ipv6ExtHeader<T>,
```

Comment:
Just curious, why are we `<T>` on `IPv6VarHeader` instead of `IPv6ExtHeader`? Seems like we'd just inherit `next_header` and `hrd_ext_len` instead of having to decorate around them in an `ext_header` field.